### PR TITLE
Documentation migrated to RST

### LINKED ISSUE #20: Change docs to rst format

### PR DESCRIPTION
**Pull request by lmikelionis**
will close #20

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+Documentation for ONGR Filter Manager Bundle can be found `here <Resources/doc/setup.rst>`_.
+
+`Read the documentation for 0.1.x (master) <https://github.com/ongr-io/FilterManagerBundle/blob/master/Resources/doc/index.rst>`_
+
+-------
+License
+-------
+
+This bundle is under the MIT license. Please, see the complete license in the bundle `LICENSE` file.
```
Comment (mindaugasbarysas) [inline README.rst]: This bundle is covered by the MIT license. Please see the complete license in the bundle `LICENSE` file.

File: Resources/doc/filter/choice.rst [hunk 2]
```diff
@@ -0,0 +1,138 @@
+=============
+Choice filter
+=============
+
+This filter groups values of repository in a specified field, and returns available options.
+If you select one of the options,  *choice filter* will return item list filtered by it.
```
Comment (mindaugasbarysas) [inline Resources/doc/filter/choice.rst]: there are two spaces before `*choice filter*`

File: Resources/doc/filter/choice.rst [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
++---------+------------+
+
+~~~~~~~~~~~~~
+Configuration
+~~~~~~~~~~~~~
+
++------------------------+--------------------------------------------------------------------------------------+
+| Setting name           | Meaning                                                                              |
++========================+======================================================================================+
+| `request_field`        | Request field used to view the selected page. (f.e. `www.page.com/?request_field=4`) |
```
Comment (mindaugasbarysas) [inline Resources/doc/filter/choice.rst]: e.g. instead of f.e.

File: Resources/doc/filter/choice.rst [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
++------------------------+--------------------------------------------------------------------------------------+
+| `field`                | Specifies the field in repository to apply this filter on. (f.e. `item_color`)       |
```
Comment (mindaugasbarysas) [inline Resources/doc/filter/choice.rst]: e.g. instead of f.e.

File: Resources/doc/filter/document_field.rst [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
++-------+-------------+
+
+~~~~~~~~~~~~~
+Configuration
+~~~~~~~~~~~~~
+
++------------------------+--------------------------------------------------------------------------------------+
+| Setting name           | Meaning                                                                              |
++========================+======================================================================================+
+| `request_field`        | Request field used to specify filter value. (f.e. `www.page.com/?request_field=3`)   |
```
Comment (mindaugasbarysas) [inline Resources/doc/filter/document_field.rst]: e.g. instead of f.e.

File: Resources/doc/filter/document_field.rst [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
++------------------------+--------------------------------------------------------------------------------------+
+| `field`                | Specifies the field in repository to apply this filter on. (f.e. `category_id`)      |
```
Comment (mindaugasbarysas) [inline Resources/doc/filter/document_field.rst]: e.g. instead of f.e.

File: Resources/doc/filter/match.rst [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+| 2       | red        |
++---------+------------+
+| 3       | blue       |
++---------+------------+
+| 4       | green      |
++---------+------------+
+| 5       | blue       |
++---------+------------+
+
+If we apply *match filter* on item_colors field with a value of `red`, we will get.
```
Comment (mindaugasbarysas) [inline Resources/doc/filter/match.rst]: we will get:

File: Resources/doc/filter/match.rst [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
++---------+------------+
+
+~~~~~~~~~~~~~
+Configuration
+~~~~~~~~~~~~~
+
++------------------------+--------------------------------------------------------------------------------------+
+| Setting name           | Meaning                                                                              |
++========================+======================================================================================+
+| `request_field`        | Request field used to specify filter value. (f.e. `www.page.com/?request_field=4`)   |
```
Comment (mindaugasbarysas) [inline Resources/doc/filter/match.rst]: e.g.

File: Resources/doc/filter/match.rst [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
++------------------------+--------------------------------------------------------------------------------------+
+| `field`                | Specifies the field in repository to apply this filter on. (f.e. `item_color`)       |
```
Comment (mindaugasbarysas) [inline Resources/doc/filter/match.rst]: e.g.

File: Resources/doc/filter/multi_choice.rst [hunk 10]
```diff
@@ -0,0 +1,124 @@
+======================
+Multiple choice filter
+======================
+
+This filter is very similar to choice filter, only you can select multiple options.
```
Comment (mindaugasbarysas) [inline Resources/doc/filter/multi_choice.rst]: "but" instead of "only"

File: Resources/doc/filter/multi_choice.rst [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+It groups values of repository in a specified field, and returns available options.
```
Comment (mindaugasbarysas) [inline Resources/doc/filter/multi_choice.rst]: It groups values of a repository in a specified field and returns available options.

File: Resources/doc/filter/multi_choice.rst [hunk 12]
```diff
@@ -0,0 +1,124 @@
...
+If you select one or more of the options,  *multi choice filter* will return item list filtered by the selected options.
```
Comment (mindaugasbarysas) [inline Resources/doc/filter/multi_choice.rst]: double space between multi choice filter

File: Resources/doc/filter/multi_choice.rst [hunk 13]
```diff
@@ -0,0 +1,124 @@
...
++=============+
++-------------+
+| red         |
++-------------+
+| green       |
++-------------+
+| blue        |
++-------------+
+
+We can then select multiple values from this list and get all items for it, lets say we select choices `blue` and `green`.
```
Comment (mindaugasbarysas) [inline Resources/doc/filter/multi_choice.rst]: let's

File: Resources/doc/filter/pager.rst [hunk 14]
```diff
@@ -0,0 +1,99 @@
...
+..
+
+~~~~~~~~~~~~~
+Configuration
+~~~~~~~~~~~~~
+
++------------------------+--------------------------------------------------------------------------------------+
+| Setting name           | Meaning                                                                              |
++========================+======================================================================================+
+| `request_field`        | Request field used to pass the page number (f.e. `www.page.com/?request_field=4`)    |
```
Comment (mindaugasbarysas) [inline Resources/doc/filter/pager.rst]: e.g.

File: Resources/doc/filter/sort.rst [hunk 15]
```diff
@@ -0,0 +1,124 @@
+===========
+Sort filter
+===========
+
+Filter used for sorting the documents.
+
+For example, lets say we have `item` repository which contains the following data:
```
Comment (mindaugasbarysas) [inline Resources/doc/filter/sort.rst]: let's

File: Resources/doc/filter/sort.rst [hunk 16]
```diff
@@ -0,0 +1,124 @@
...
+| 2       | red        |
++---------+------------+
+| 3       | blue       |
++---------+------------+
+| 4       | green      |
++---------+------------+
+| 5       | blue       |
++---------+------------+
+
+If we have a sort filter with choice to sort on `item_color` in descending order and this choice is selected we will get:
```
Comment (mindaugasbarysas) [inline Resources/doc/filter/sort.rst]: If we have a sort filter with choice to sort on `item_color` in descending order and this choice is selected, we will get:

File: Resources/doc/filter/sort.rst [hunk 17]
```diff
@@ -0,0 +1,124 @@
...
+~~~~~~~~~~~~~
+Configuration
+~~~~~~~~~~~~~
+
+First, you have to specify the request field:
+
++------------------------+--------------------------------------------------------------------------------------+
+| Setting name           | Meaning                                                                              |
++========================+======================================================================================+
+| `request_field`        | Request field used to pass the sort choice id (f.e. `www.page.com/?request_field=4`) |
```
Comment (mindaugasbarysas) [inline Resources/doc/filter/sort.rst]: s/f\.e\./e\.g\./g

File: Resources/doc/filter/sort.rst [hunk 18]
```diff
@@ -0,0 +1,124 @@
...
++------------------------+--------------------------------------------------------------------------------------+
+
+After which you can specify multiple sort options/choices:
+
++------------------------+--------------------------------------------------------------------+
+| Setting name           | Meaning                                                            |
++========================+====================================================================+
+| `label`                | Choice name to be used in templates. (f.e. `Title descending`)     |
```
Comment (mindaugasbarysas) [inline Resources/doc/filter/sort.rst]: e.g.

File: Resources/doc/filter/sort.rst [hunk 19]
```diff
@@ -0,0 +1,124 @@
...
++------------------------+--------------------------------------------------------------------+
+| `field`                | Specifies the field in repository to sort on. (f.e. `item_color`)  |
```
Comment (mindaugasbarysas) [inline Resources/doc/filter/sort.rst]: e.g.

File: Resources/doc/index.rst [hunk 20]
```diff
@@ -0,0 +1,22 @@
...
+
+Documentation for ONGR Filter Manager Bundle can be found `here <setup.rst>`_.
+
+`Read the documentation for 0.1.x (master) <https://github.com/ongr-io/FilterManagerBundle/blob/master/Resources/doc/index.rst>`_
+
+-------
+License
+-------
+
+This bundle is under the MIT license. Please, see the complete license in the bundle `LICENSE` file.
```
Comment (mindaugasbarysas) [inline Resources/doc/index.rst]: This bundle is covered by the MIT license. Please see the complete license in the bundle `LICENSE` file.

File: Resources/doc/manager.rst [hunk 21]
```diff
@@ -0,0 +1,97 @@
...
+
+
+FilterManager is used for retrieving filtered entities from database. It provides ties between commonly used filtering options and UI elements with Elasticsearch repositories.
+You can use it from a single controller.
+
+It requires *managers* and *filters* configured to work.
+
+*Managers* are used to specify repository which will be filtered and which *filters* will be available for the given repository.
+
+*Filters* are a set of rules, by which the documents will be filtered.
```
Comment (mindaugasbarysas) [inline Resources/doc/manager.rst]: *Filters* are a set of rules by which the documents will be filtered.

File: Resources/doc/manager.rst [hunk 22]
```diff
@@ -0,0 +1,97 @@
...
+                item_pager:
+                    request_field: 'page'
+                    count_per_page: 9
+                category_pager:
+                    request_field: 'category_list_page'
+                    count_per_page: 3
+
+..
+
+This configuration would create six filters, which would be used by a single or both managers, the resulting structure can be seen as a table:
```
Comment (mindaugasbarysas) [inline Resources/doc/manager.rst]: This configuration would create six filters, which would be used by a single or both managers. The resulting structure can be seen as a table:

File: Resources/doc/usage.rst [hunk 23]
```diff
@@ -0,0 +1,136 @@
...
+        pattern: /list
+        methods:  [GET]
+        defaults:
+            _controller: AcmeDemoBundle:List:index
+
+..
+
+.. code-block:: php
+
+    //src/Acme/DemoBundle/Controller/ListController.php
```
Comment (mindaugasbarysas) [inline Resources/doc/usage.rst]: this doesn't look right. Would it pass our ONGR-strict-standard? ;)

File: Resources/doc/usage.rst [hunk 24]
```diff
@@ -0,0 +1,136 @@
...
+        /**
+         * Returns item list
+         *
+         * @param Request $request
+         *
+         * @return array
+         */
+        private function getProductsData($request)
+        {
+            //here we get our filter manager
```
Comment (mindaugasbarysas) [inline Resources/doc/usage.rst]: This wouldn't pass our standard as well.

File: Resources/doc/usage.rst (hunk 23 above)
Comment (lmikelionis) [inline Resources/doc/usage.rst]: you are talking about comment?

Comment (mindaugasbarysas) [inline Resources/doc/usage.rst]: Yes, it  should probably read something like this:
` // File location: src/.../ListController.php.`

**Pull request merged**